File: app/src/main/java/com/syh/framework/list/ListActivity.java
package com.syh.framework.list;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.widget.SwipeRefreshLayout;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;

import com.syh.framework.R;
import com.syh.framework.util.BaseActivity;

import java.util.ArrayList;
import java.util.List;

/**
 * Created bg shenyonghe on 2018/6/7.
 */
public class ListActivity extends BaseActivity {

    private RecyclerView recyclerView;
    private SwipeRefreshLayout refreshLayout;
    private MyQuickAdapter adapter;
    private int num = 1;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_list);
        initRV();
    }

    private void initRV() {
        recyclerView = findViewById(R.id.recycler_view);
        refreshLayout = findViewById(R.id.refreshLay);
        refreshLayout.setOnRefreshListener(() -> {
            refreshList();
        });
        recyclerView.setLayoutManager(new LinearLayoutManager(this));
        adapter = new MyQuickAdapter(getData());
        recyclerView.setAdapter(adapter);
    }

    private List<ItemDemo> getData() {
        List<ItemDemo> list = new ArrayList<>();
        for (int i = 0; i < 20; i++) {
            ItemDemo itemDemo = new ItemDemo();
            itemDemo.setName("shen" + i);
            itemDemo.setKeyNum(i + "key");
            list.add(itemDemo);
        }
        return list;
    }


    private List<ItemDemo> getRefreshData() {
        List<ItemDemo> list = getData();
        list.remove(4);
        list.remove(5);
        list.remove(6);
        return list;
    }

    private void refreshList() {
        adapter.setDatas(getRefreshData());
        refreshLayout.setRefreshing(false);
    }
}

File: app/src/main/java/com/syh/framework/list/ListActivity.kt
package com.syh.framework.list

import android.os.Bundle
import android.support.v4.widget.SwipeRefreshLayout.OnRefreshListener
import android.support.v7.widget.LinearLayoutManager
import com.syh.framework.R
import com.syh.framework.expose.ExposeManager
import com.syh.framework.util.BaseActivity
import com.syh.plugin.annotation.Cost
import kotlinx.android.synthetic.main.activity_list.*
import java.util.*

/**
 * Created bg shenyonghe on 2018/6/7.
 */
class ListActivity : BaseActivity() {
    private var adapter: MyQuickAdapter? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_list)
        initRV()
        ExposeManager.report(this)
    }

    @Cost
    private fun initRV() {
        refreshLay.setOnRefreshListener(OnRefreshListener { refreshList() })
        recycler_view.layoutManager = LinearLayoutManager(this)
        adapter = MyQuickAdapter(data)
        recycler_view.adapter = adapter
    }

    private val data: MutableList<ItemDemo?>
        private get() {
            val list: MutableList<ItemDemo?> = ArrayList()
            for (i in 0..19) {
                val itemDemo = ItemDemo()
                itemDemo.name = "shen$i"
                itemDemo.keyNum = i.toString() + "key"
                list.add(itemDemo)
            }
            return list
        }
    private val refreshData: MutableList<ItemDemo?>
        private get() {
            val list = data
            list.removeAt(4)
            list.removeAt(5)
            list.removeAt(6)
            return list
        }

    private fun refreshList() {
        adapter!!.setDatas(refreshData)
        refreshLay!!.isRefreshing = false
    }
}